packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -1057,23 +1033,6 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
       await idle();
     });
   }
-
-  /// Makes an effort to dismiss the current page with a Material [Scaffold] or
-  /// a [CupertinoPageScaffold].
-  ///
-  /// Will throw an error if there is no back button in the page.
-  Future<void> pageBack() async {
-    return TestAsyncUtils.guard<void>(() async {
-      Finder backButton = find.byTooltip('Back');
-      if (backButton.evaluate().isEmpty) {
-        backButton = find.byType(CupertinoNavigationBarBackButton);
-      }
-
-      expectSync(backButton, findsOneWidget, reason: 'One back button expected on screen');
```

Comment:
Throwing an exception when there's no back button still seems to be desired. As `expectSync` may not be available from `WidgetController`, maybe we can use something else (e.g., [FlutterError]). That way, we can probably still preserve the `fails when there are no back buttons` test with a small modification.

[FlutterError]: https://api.flutter.dev/flutter/foundation/FlutterError-class.html

Reply:
I wonder if it really is desired/consistent with other APIs. By removing the `pageBack` method, IDE reminds me that the `matcher.dart` is no longer needed: basically this method is the only method `expect`ing the existence of a finder, while all other methods are assuming a finder exists, and rely on when really evaluating a finder to raise an error.

Reply:
this whole method is a bit dubious to be honest (we really shouldn't be depending on material or cupertino in flutter_test). It's all just convenience methods. So once we have them anyway, why not also make sure there's a back button.

That said, whether we throw a FlutterError or a TestError (or whatever expectSync fires) doesn't seem to matter that much. This is a test framework, so expectSync is not unreasonable IMHO.

Reply:
as it is you're still going to through when you try to tap a the button and there isn't one